toptal/picasso
# [FX-4812] Create Paper component

[FX-4812]

This pull request migrates Paper component to Tailwind, so it stops relying on Material UI.

### How Paper relied on Material UI?

1. Shadows – Material UI provided 25 shadows (`elevation` property of Paper component). First 6 shadows are exactly the same as the ones in BASE, the remaining ones will be deprecated and removed in https://toptal-core.atlassian.net/browse/FX-5003. Picasso consumers use different values (`0`, `1`, `2`, `3`, `6`, `15` were found via GitHub search), so this will be a breaking change carried out separately. The goal for now is to seamlessly migrate to Tailwind.

2. `MUIPaper` component – no longer used, Paper uses plain `<div/>` instead

3. Styling solution from Material UI (`makeStyles`, `createStyles`) – no longer used, Paper appends `className` property to the list of Tailwind classes

### Why Modals, Drawer, and Popover are not covered?

These components use `<MUIPaper/>` under the hood, so they will start using the Paper from this pull request when they will be migrated to Tailwind themselves.

### How to test


- [Temploy](https://picasso.toptal.net/fx-4812-create-paper-component)
- All checks should pass
- Check visual changes in https://github.com/toptal/client-portal/pull/8332 (there are 7 deleted screenshots, but this is coming from `master`)

### Development checks


**Breaking change**

- [N/A] codemod is created and showcased in the changeset
- [x] test alpha package of Picasso in Client Portal https://github.com/toptal/client-portal/pull/8332

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
3. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
5. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[FX-4812]: https://toptal-core.atlassian.net/browse/FX-4812?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
